statsd/statsd.go

```diff
@@ -58,3 +60,15 @@ func (s *Statsd) EventsFromKinesis(num int, shardID string, lag time.Duration) {
 	_ = s.client.TimingDuration(fmt.Sprintf("kinsumer.%s.lag", shardID), lag, 1.0)
 	_ = s.client.Inc(fmt.Sprintf("kinsumer.%s.retrieved", shardID), int64(num), 1.0)
 }
+
+// RecordsInMemory implementation that writes to statsd a gauge metric about
+// the current number of records buffered in memory
+func (s *Statsd) RecordsInMemory(count int) {
+	_ = s.client.Gauge("kinsumer.records_in_memory", int64(count), 1.0)
+}
+
+// RecordsInMemoryBytes implementation that writes to statsd a gauge metric about
+// the current total payload bytes buffered in memory
+func (s *Statsd) RecordsInMemoryBytes(bytes int64) {
+	_ = s.client.Gauge("kinsumer.records_in_memory_bytes", bytes, 1.0)
```

Comment:
If you want these exported out to cloudwatch the metrics need to be prefixed with "snowplow.<something>" - not sure how we are currently dealing with the kinsumer metrics today?